pkg/util/provider/machinecontroller/machine.go

```diff
@@ -585,6 +600,55 @@ func (c *controller) triggerCreationFlow(ctx context.Context, createMachineReque
 	return machineutils.LongRetry, nil
 }
 
+func (c *controller) initializeMachine(ctx context.Context, machine *v1alpha1.Machine, machineClass *v1alpha1.MachineClass, secret *corev1.Secret) (machineutils.RetryPeriod, error) {
+	req := &driver.InitializeMachineRequest{
+		Machine:      machine,
+		MachineClass: machineClass,
+		Secret:       secret,
+	}
+	klog.V(3).Infof("Initializing VM instance for Machine %q", machine.Name)
+	resp, err := c.driver.InitializeMachine(ctx, req)
+	if err != nil {
+		errStatus, ok := status.FromError(err)
+		if !ok {
+			klog.Errorf("Error occurred while decoding machine error for machine %q: %s", machine.Name, err)
+			return machineutils.MediumRetry, err
+		}
+		klog.Errorf("Error occurred while initializing VM instance for machine %q: %s", machine.Name, err)
+		switch errStatus.Code() {
+		case codes.Unimplemented:
+			klog.V(3).Infof("Provider does not support Driver.InitializeMachine - skipping VM instance initialization for %q.", machine.Name)
+			return 0, nil
+		case codes.NotFound:
+			klog.Warningf("No VM instance found for machine %q. Skipping VM instance initialization.", machine.Name)
+			return 0, nil
+		case codes.Canceled, codes.Aborted:
```

Comment:
Sorry for late response. If it is not found/aborted, there is nothing to initialize. No need to do any special logic since our current code in MCM will take care of removing the machine after some time. I can add a test case for this. (Actually, I should explicitly log this test result - will do)

Reply:
Can we remove the case for `Aborted` and `Canceled` as it is unclear when the driver will return these error codes. We don't see usage of these codes in the driver implementation.

Reply:
I can remove the code, but after reading `docs/development/machine_error_codes.md` for other operations, the spec says that `CANCELED` should be handled with a retry.   If I remove both codes, then Cancelled will be retried. Is this fine ?

Reply:
In addition, according to spec, `INVALID_ARGUMENT`  must not be re-tried...should I change to adopt the spec ?

Reply:
Ok confirmed that majority of error codes can be ignored as this is legacy behaviour in the document. we need to clean up and fix all this bit-rot in the overhaul.